lib/plugins/aws/utils/request.js

```diff
@@ -0,0 +1,208 @@
+  const caPathArr = cafile.split(',');
+  caCerts = caCerts.concat(caPathArr.map((cafilePath) => fs.readFileSync(cafilePath.trim())));
+}
+
+if (caCerts.length > 0) {
+  Object.assign(proxyOptions, {
+    rejectUnauthorized: true,
+    ca: caCerts,
+  });
+}
+
+// Passes also certifications
+if (proxy) {
+  sdk.config.httpOptions.agent = new HttpsProxyAgent(proxyOptions);
+} else if (proxyOptions.ca) {
+  // Update the agent -- http://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/node-registering-certs.html
+  sdk.config.httpOptions.agent = new https.Agent(proxyOptions);
+}
+
+// Configure the AWS Client timeout (Optional).  The default is 120000 (2 minutes)
+const timeout = process.env.AWS_CLIENT_TIMEOUT || process.env.aws_client_timeout;
+if (timeout) {
+  sdk.config.httpOptions.timeout = parseInt(timeout, 10);
+}
+PromiseQueue.configure(BbPromise.Promise);
+const requestQueue = new PromiseQueue(2, Infinity);
+
+const MAX_RETRIES = (() => {
+  const userValue = Number(process.env.SLS_AWS_REQUEST_MAX_RETRIES);
+  return userValue >= 0 ? userValue : 4;
+})();
+
+function awsRequest(service, method, params = {}, serviceOptions = {}) {
+  // Checks regarding expectations on serviceOptions
+  if (!_.isObject(serviceOptions) || !_.isObject(serviceOptions.credentials)) {
+    return BbPromise.reject(
+      new ServerlessError(
+        'Inappropriate call of awsRequest(), missing credentials in serviceOptions',
+        400
+      )
+    );
+  }
+
+  const BASE_BACKOFF = 5000;
+  const persistentRequest = (f) =>
+    new BbPromise((resolve, reject) => {
+      const doCall = (numTry) => {
+        f()
+          // We're resembling if/else logic, therefore single `then` instead of `then`/`catch` pair
+          .then(resolve, (e) => {
+            const { providerError } = e;
+            if (
+              numTry < MAX_RETRIES &&
+              providerError &&
+              ((providerError.retryable &&
+                providerError.statusCode !== 403 &&
+                providerError.code !== 'CredentialsError') ||
+                providerError.statusCode === 429)
+            ) {
+              const nextTryNum = numTry + 1;
+              const jitter = Math.random() * 3000 - 1000;
+              // backoff is between 4 and 7 seconds
+              const backOff = BASE_BACKOFF + jitter;
+
+              log(
+                [
+                  `Recoverable error occurred (${e.message}), sleeping for ~${Math.round(
+                    backOff / 1000
+                  )} seconds.`,
+                  `Try ${nextTryNum} of ${MAX_RETRIES}`,
+                ].join(' ')
+              );
+              setTimeout(doCall, backOff, nextTryNum);
+            } else {
+              reject(e);
+            }
+          });
+      };
+      return doCall(0);
+    });
+
+  // Support S3 Transfer Acceleration
+  if (
+    canUseS3TransferAcceleration(service, method) &&
```

Comment:
Let's remove this function, and put this resolution inline (it's one time used one liner, I don't think it's justified to have it constructed with extra function)